--- a/config/ymake	2019-03-06 18:04:12.178585855 -0500
+++ b/config/ymake	2019-03-06 18:04:57.766739265 -0500
@@ -535,6 +535,8 @@
         breaksw
 endsw
 
+set sysincs = Spack
+
 if ($?sysincs == 0) then
         echo "$0 : Unknown System Type - No Config file" > /dev/tty
         exit 1
--- a/Configure	2019-03-06 17:56:35.497074727 -0500
+++ b/Configure	2019-03-06 17:59:21.870341768 -0500
@@ -1114,13 +1114,13 @@
   set libs = ($loclibdir:x)
   set newloclib=""
   while ($#libs > 0)
-    if (! -d $libs[1]) then
-      echo "  *** Warning: <$libs[1]> does not exist"
-      echo ""
-      goto proc_loclibdir
-    else 
+    # if (! -d $libs[1]) then
+    #   echo "  *** Warning: <$libs[1]> does not exist"
+    #   echo ""
+    #   goto proc_loclibdir
+    # else 
       set newloclib = ($newloclib "-L$libs[1]")
-    endif
+    # endif
     shift libs
   end
   set loclib = "$newloclib"
@@ -1164,13 +1164,13 @@
   set incs = ($locincdir:x)
   set newlocinc=""
   while ($#incs > 0)
-    if (! -d $incs[1]) then
-      echo "  *** Warning: <$incs[1]> does not exist"
-      echo ""
-      goto proc_locincdir
-    else 
+    # if (! -d $incs[1]) then
+    #   echo "  *** Warning: <$incs[1]> does not exist"
+    #   echo ""
+    #   goto proc_locincdir
+    # else 
       set newlocinc = ($newlocinc "-I$incs[1]")
-    endif
+    # endif
     shift incs
   end
   set locinc = "$newlocinc"
